pubsub/subscriptions/main.go

```
@@ -88,12 +87,19 @@ func pullMsgs(client *pubsub.Client, name string, topic *pubsub.Topic) error {
 	ctx := context.Background()
 
 	// publish 10 messages on the topic.
+	var results []*pubsub.PublishResult
 	for i := 0; i < 10; i++ {
-		_, err := topic.Publish(ctx, &pubsub.Message{
+		res := topic.Publish(ctx, &pubsub.Message{
 			Data: []byte(fmt.Sprintf("hello world #%d", i)),
 		})
+		results = append(results, res)
+	}
+
+	// Check if all messages are published.
+	for _, r := range results {
+		_, err := r.Get(ctx)
 		if err != nil {
-			return err
+			log.Printf("Publish error: %v", err)
```

This should return the error